src/core/mount.c

```diff
@@ -1854,6 +1887,42 @@ static int mount_setup_unit(
         return 0;
 }
 
+static int mount_load_proc_self_device(Manager *m) {
+        _cleanup_(mnt_free_tablep) struct libmnt_table *table = NULL;
+        _cleanup_(mnt_free_iterp) struct libmnt_iter *iter = NULL;
+        MountDeviceNode *head = NULL;
+        int r;
+
+        assert(m);
+
+        r = libmount_parse(NULL, NULL, &table, &iter);
```

Comment:
uh, this is racy, because the mount table will now be read twice, and it might have changed in between, but some operations have done based on the first reading, and the other on the second reading.